### ISSUE TITLE
how to request.respond() multiple set-cookie headers?

### ISSUE THREAD

**Issue by ghost** [puppeteer/puppeteer#5740]
If I look at the respond code int NetworkManager.js it only accepts a single 'set-cookie' header as a string.

```javascript
  async respond(response) {
    /** @type {!Object<string, string>} */
    const responseHeaders = {};
    if (response.headers) {
      for (const header of Object.keys(response.headers))
        responseHeaders[header.toLowerCase()] = response.headers[header];
    }
    if (response.contentType)
      responseHeaders['content-type'] = response.contentType;
    if (responseBody && !('content-length' in responseHeaders))
      responseHeaders['content-length'] = String(Buffer.byteLength(responseBody));
```

I've tried
```
{headers: {'set-cookie': '1=1.;Domain=.site.com;Path=/;Expires=Tue, 26-Apr-2022 10:30:53 GMT;Priority=HIGH,2=2.;Domain=.site.com;Path=/;Expires=Tue, 26-Apr-2022 10:30:53 GMT;Secure;HttpOnly;Priority=HIGH;SameSite=none,3=3.;Path=/;Expires=Tue, 26-Apr-2022 10:30:53 GMT;Secure;HttpOnly;Priority=HIGH'}}
```
and
```
{headers: {'set-cookie': '1=1.;Domain=.site.com;Path=/;Expires=Tue, 26-Apr-2022 10:30:53 GMT;Priority=HIGH, 2=2.;Domain=.site.com;Path=/;Expires=Tue, 26-Apr-2022 10:30:53 GMT;Secure;HttpOnly;Priority=HIGH;SameSite=none, 3=3.;Path=/;Expires=Tue, 26-Apr-2022 10:30:53 GMT;Secure;HttpOnly;Priority=HIGH'}}
```
but neither set the cookies

If I mock `request.respond` and put the set-cookie each on their own header it works.

**Comment by Ganitzsh** [comment]
I have a similar issue, did you find a solution to this?

**Comment by ghost** [comment]
> I have a similar issue, did you find a solution to this?

I think I mocked request.respond with my own implementation. it's been a while, don't have the code anymore.

**Comment by FranckFreiburger** [comment]
Maybe this little source code modification can do the trick:
https://github.com/puppeteer/puppeteer/blob/62ac16733913b388e2a4904ff6ef21aa431475e9/src/common/HTTPRequest.ts#L380

`responseHeaders: Array.isArray(responseHeaders) ? responseHeaders : headersArray(responseHeaders),`

**Comment by onlyil** [comment]
I have a similar issue, did you find a solution to this?

**Comment by dukei** [comment]
Any progress on this issue? It is still not possible to pass multiple set-cookie headers to request.respond().

**Comment by Junyan** [comment]
@dukei @onlyil @dukei The issue is solved, see [v13.6.0](https://github.com/puppeteer/puppeteer/releases/tag/v13.6.0).
